google/cloud/sql/connector/instance.py

```diff
@@ -358,6 +358,14 @@ async def _perform_refresh(self) -> InstanceMetadata:
                 if expiration > token_expiration:
                     expiration = token_expiration
 
+        except aiohttp.ClientResponseError as e:
+            logger.debug(
+                f"['{self._instance_connection_string}']: Error occurred during _perform_refresh."
+            )
+            if e.code == 403:
+                e.message = "Forbidden: Authenticated IAM principal does not seeem authorized to make API request. Verify 'Cloud SQL Client' role has been granted to IAM principal."
+            raise
+
```

Comment:
Will add test for this to my other test PR if this gets merged first. If other test PR gets merged first then I can add test to this PR.